remindme-api/src/main/java/it/andreafailli/remindme/api/auth/FirebaseAuthenticationProvider.java

```diff
@@ -0,0 +1,49 @@
+package it.andreafailli.remindme.api.auth;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.AuthorityUtils;
+import org.springframework.stereotype.Component;
+
+import it.andreafailli.remindme.common.models.User;
+import it.andreafailli.remindme.common.services.UserService;
+
+@Component
+public class FirebaseAuthenticationProvider implements AuthenticationProvider {
+
+    @Autowired
+    private UserService userService;
+
+    public boolean supports(Class<?> authentication) {
+        return (FirebaseAuthenticationToken.class.isAssignableFrom(authentication));
+    }
+
+    public Authentication authenticate(Authentication authentication) throws AuthenticationException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.springframework.security.core.AuthenticationException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)